### ISSUE TITLE
issue with auto_dc_coords

### ISSUE THREAD

**Issue by AnoopAlias** [gdnsd/gdnsd#173]
Today, I have failed gdnsd because the auto_dc_coords complains of a lack of latitude and longitude data

```
gdnsd checkconf
# info: Loading configuration from '/etc/gdnsd/config'
# info: DNS listener threads (1 UDP + 1 TCP) configured for 0.0.0.0:53
# info: DNS listener threads (1 UDP + 1 TCP) configured for [::]:53
# warning: The default value of the global option 'zones_rfc1035_auto' will likely change from 'true' to 'false' in a future version.  Setting the value explicitly for forward-compatibility is recommended!
# info: plugin_geoip: map 'auto_map': Loading GeoIP2 database '/etc/gdnsd/geoip/GeoLite2-City.mmdb': Version: 2.0, Type: GeoLite2-City, IPVersion: 6, Timestamp: 2019-01-04 00:03:13 UTC
# error: plugin_geoip: map 'auto_map': GeoIP2 DB '/etc/gdnsd/geoip/GeoLite2-City.mmdb' appears to be a post-2018 GeoLite2-City database, which will not work with auto_dc_coords as configured because these databases lack the latitude and longitude data present in the commercial version.  See the auto_dc_coords section of the gdnsd-plugin-geoip documentation for more details.
# error: plugin_geoip: map 'auto_map': (Re-)loading geoip database '/etc/gdnsd/geoip/GeoLite2-City.mmdb' failed!
# fatal: plugin_geoip: map 'auto_map': cannot continue initial load
```

But as per Maxmind the lat/long will not be removed from the free geolite2 

https://blog.maxmind.com/2018/04/11/we-are-listening-latitude-and-longitude-will-remain-in-geolite2-databases/

What am I missing?

**Comment by AnoopAlias** [comment]
From what I see the code is just checking the timestamp and refusing to load?

**Comment by AnoopAlias** [comment]
Set the date check in  https://github.com/gdnsd/gdnsd/blob/master/libgdmaps/gdgeoip2.c#L171

to a future date and gdnsd loads just fine the .mmdb

I guess you need to remove that if since MaxMind doesn't remove the lat/long info from their GeoLite2

**Comment by blblack** [comment]
Yeah, I hardcoded that date back when I saw the original announcements, since the loss of coordinate data was going to break things for users, but I never heard about the retraction, sorry!

A patch to the master branch is incoming shortly, and I'll do a backport as 2.4.1 as well.

**Comment by blblack** [comment]
So the fix is released with [v2.4.1](https://github.com/gdnsd/gdnsd/releases/tag/v2.4.1) stable release, and it's in master for the next 2.99.x beta release as well (which there shouldn't be too many more of, I just have a little more testing and QA work to do there, and possibly one more minor feature change).  I'll close this up once the next beta is out the door as well.  Thanks for reporting!

**Comment by blblack** [comment]
Fix was released with gdnsd-2.99.9950-beta
